db/migrations/sqls/20220427152150-ddc-metric-up.sql

```diff
@@ -0,0 +1,19 @@
+CREATE TABLE IF NOT EXISTS ddc_metric
+(
+    id     SERIAL NOT NULL,
+    name   TEXT   NOT NULL,
+    nodeId BIGINT,
+    PRIMARY KEY (name, nodeId)
+);
+
+CREATE TABLE IF NOT EXISTS ddc_metric_value
+(
+    metricId  BIGINT         NOT NULL,
+    value     NUMERIC(40, 0) NOT NULL,
+    timestamp BIGINT         NOT NULL
+);
+
+CREATE OR REPLACE VIEW ddc_metric_view AS
+SELECT name, nodeId, value, timestamp
+FROM ddc_metric_value
+         JOIN ddc_metric on ddc_metric_value.metricId = ddc_metric.id;
```

Comment:
Why do we use a `view` here?

Reply:
Graphql doesn't provide joins.  Can we transform graphql request to database join query? Different clients may require different data and it doesn't make much sense to create views for each specific client request.

Reply:
graphql can send a requests to views as to regular tables. It's exactly why we need this view.

You can think about this view as a denormalised representation of metrics. This view is used by all 12 metrics. No need to write view per each specific one.

Reply:
yes

Reply:
If we don't need to write any additional code in Hasura according to this view then it's great.